modules/@apostrophecms/piece-type/index.js

```diff
@@ -388,20 +395,61 @@ module.exports = {
       },
       'apostrophe:modulesRegistered': {
         composeBatchOperations() {
-          self.batchOperations = Object.keys(self.batchOperations).map(key => ({
-            action: key,
-            ...self.batchOperations[key]
-          })).filter(batchOperation => {
-            // If a `requiredField` is registered, only include the operation
-            // if that field is present on the schema.
-            if (batchOperation.requiredField && !_.find(self.schema, {
-              name: batchOperation.requiredField
-            })) {
-              return false;
+          const groupedOperations = Object.entries(self.batchOperations)
+            .reduce((acc, [ opeName, properties ]) => {
+              const requiredFieldNotFound = properties.requiredField && !self.schema
+                .some((field) => field.name === properties.requiredField);
+
+              if (requiredFieldNotFound) {
+                return acc;
+              }
+
+              const associatedGroup = getAssociatedGroup(opeName);
+              const currentOperation = {
+                name: opeName,
+                ...properties
+              };
+              const { name, ...props } = getCurrentOrGroup(
+                currentOperation,
+                associatedGroup,
+                acc
+              );
+
+              return {
+                ...acc,
+                [name]: {
+                  ...props
+                }
+              };
+            }, {});
+
+          self.batchOperations = Object.entries(groupedOperations)
+            .map(([ name, properties ]) => ({
+              name,
+              ...properties
+            }));
+
+          function getCurrentOrGroup (currentOpe, [ groupName, groupProperties ], acc) {
+            if (!groupName) {
+              return currentOpe;
             }
 
-            return true;
-          });
+            return {
+              name: groupName,
+              ...groupProperties,
+              operations: [
+                ...acc[groupName]?.operations || [],
+                currentOpe
+              ]
+            };
+          }
+
+          function getAssociatedGroup (operation) {
+            return Object.entries(self.batchOperationsGroups)
```

Comment:
I don't see `batchOperationsGroups` anywhere else. I'm not really clear on what is happening in this other than eventually returning the empty array. The use of `_` is throwing me as well since it is assigned to lodash above.

Reply:
I didn't dive really deep into but it seems that it's created by the cascading system, it automatically computes operations that are in groups... I guess it works the same way for all the cascading properties. 
I will rename the `_`, I use this convention for param that are not used (eslint doesn't throw error). I renamed it to `_key`.

Reply:
Oh, that's right. It's built with a template literal so the specific term isn't anywhere else.